### PR TITLE
Enable dylib-llvm-aot backend e2e/xla_op tests

### PR DESCRIPTION
**Pull request by asaadaldien**
Enable dylib-llvm-aot xla_op tests.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -115,7 +115,7 @@ endforeach()
 
 # List of all target backends to be built by default:
 set(IREE_ALL_TARGET_BACKENDS
-  # TODO(scotttodd): LLVMAOT
+  DYLIB-LLVM-AOT
```
Comment (ScottTodd) [inline CMakeLists.txt:118]: could also (future PR?) update some parts of https://github.com/google/iree/blob/main/scripts/update_op_coverage.py and https://github.com/google/iree/blob/main/scripts/update_e2e_coverage.py

(may also need style changes to make the tables accommodate an extra column)

File: iree/modules/check/CMakeLists.txt [hunk 2]
```diff
@@ -57,6 +57,7 @@ iree_cc_binary(
     iree::base::status
     iree::base::target_platform
     iree::base::tracing
+    iree::hal::dylib::dylib_driver_module
```
Comment (ScottTodd) [inline iree/modules/check/CMakeLists.txt:60]: missing a corresponding change to `iree/modules/check/BUILD`?

Comment (asaadaldien) [reply iree/modules/check/CMakeLists.txt:60]: Its in IREE_DRIVER_MODULES list in `bazel_to_cmake_converter.py` which is updated in this PR. CI didn't run it !!?

Comment (ScottTodd) [reply iree/modules/check/CMakeLists.txt:60]: Huh, weird. Let's re-run the GitHub actions checks to see if it's a transient error. This does look correct on closer inspection...

Comment (ScottTodd) [reply iree/modules/check/CMakeLists.txt:60]: Still shows the same diff. @GMNGeoffrey any ideas?

File: build_tools/bazel/build_core.sh [hunk 3]
```diff
@@ -42,6 +42,7 @@ fi
 declare -a test_env_args=(
   --test_env=IREE_LLVMJIT_DISABLE=$IREE_LLVMJIT_DISABLE
   --test_env=IREE_VULKAN_DISABLE=$IREE_VULKAN_DISABLE
+  --action_env=IREE_LLVMAOT_LINKER_PATH=$IREE_LLVMAOT_LINKER_PATH
```
Comment (GMNGeoffrey) [inline build_tools/bazel/build_core.sh:45]: Please add this to the env for all the bazel builds, most importantly the bindings build, which also uses RBE. The environment is part of the cache key, which means we want this to be the same everywhere to get cache hits.

File: iree/modules/check/CMakeLists.txt (hunk 2 above)
Comment (GMNGeoffrey) [reply iree/modules/check/CMakeLists.txt:60]: weird... Let me try fiddling with it locally

Comment (GMNGeoffrey) [reply iree/modules/check/CMakeLists.txt:60]: Ok the issue is that the workflow is checking out the merge commit for the action which somehow does not have your bazel_to_cmake change. https://github.com/google/iree/commit/89dd7e42866c334f89b57def2605b274d9d16ca6 I don't understand why though. My recommendation would be to rebase and force push and see if that fixes it.

File: CMakeLists.txt (hunk 1 above)
Comment (asaadaldien) [reply CMakeLists.txt:118]: https://github.com/google/iree/issues/2673

File: iree/modules/check/CMakeLists.txt (hunk 2 above)
Comment (asaadaldien) [reply iree/modules/check/CMakeLists.txt:60]: Rebased, it works now.

**Comment by ScottTodd** [comment]
Please re-request review after addressing comments so your PR shows up under https://github.com/google/iree/pulls/review-requested/@me

File: build_tools/cmake/iree_check_test.cmake [hunk 4]
```diff
@@ -243,8 +243,8 @@ function(iree_check_test_suite)
   )
 
   if(NOT DEFINED _RULE_TARGET_BACKENDS AND NOT DEFINED _RULE_DRIVERS)
-    set(_RULE_TARGET_BACKENDS "vmla" "vulkan-spirv" "llvm-ir")
-    set(_RULE_DRIVERS "vmla" "vulkan" "llvm")
+    set(_RULE_TARGET_BACKENDS "vmla" "vulkan-spirv" "llvm-ir" "dylib-llvm-aot")
+    set(_RULE_DRIVERS "vmla" "vulkan" "llvm" "dylib")
```
Comment (GMNGeoffrey) [inline build_tools/cmake/iree_check_test.cmake:247]: This change should be mirrored at https://github.com/google/iree/blob/main/build_tools/bazel/iree_check_test.bzl#L20-L24, right?

**Comment by ScottTodd** [comment]
Whoops, this now causes developer builds to fail if they don't specify `IREE_LLVMAOT_LINKER_PATH`. Can we make this fail more gracefully, work by default, or at least update documentation across all platforms/configurations? Relates to https://github.com/google/iree/issues/2645.

**Pull request merged**